nav2_ceres_costaware_smoother/include/nav2_ceres_costaware_smoother/utils.hpp

```diff
@@ -0,0 +1,93 @@
+// Copyright (c) 2021 RoboTech Vision
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef NAV2_CERES_COSTAWARE_SMOOTHER__UTILS_HPP_
+#define NAV2_CERES_COSTAWARE_SMOOTHER__UTILS_HPP_
+
+#include <limits>
+#include "Eigen/Core"
+
+#define EPSILON 0.0001
+
+namespace nav2_ceres_costaware_smoother
+{
+
+template<typename T>
+inline int sign(T a)
+{
+  return a == 0 ? 0 : a > 0 ? 1 : -1;
+}
+
+template<typename T>
+inline Eigen::Matrix<T, 2, 1> arcCenter(
+  Eigen::Matrix<T, 2, 1> p1,
```

Comment:
Naming these variables a little more meaningfully would help alot in the understanding of the code. I think this is point i+1, point i, and point i - 1, but the actual names don't make that clear